common/ast/Cargo.toml

```diff
@@ -8,12 +8,12 @@ edition = "2021"
 
 [lib]
 doctest = false
-test = false
```

Comment:
Did we intended to disable the unit tests?

Reply:
Hi, databend's unit tests are built too slow, so we move all of them to integration tests instead. Please take a test at https://github.com/datafuselabs/databend/discussions/3473

Reply:
Currently, we do not have any so-called unit tests :)

Reply:
@PsiACE @Xuanwo Ah, thanks for your fast response. So I'm curious about how do you test the private interface.

Reply:
> @PsiACE @Xuanwo Ah, thanks for your fast response. So I'm curious about how do you test the private interfaces. In addition, how do you test with some mock technics with `#[cfg(test)]`.

By making it pub 😆

Reply:
Two methods: 
1. add the test to src/tests , remove test = false.
2. Make it public for now and put it under tests/it.

The vast majority of tests in Databend follow the 2nd method. 

We are exploring new ways to balance the testing needs of private modules.

Reply:
Oh, then the fabulous mocking tool [mockall](https://crates.io/crates/mockall) will not work...

Reply:
@andylokandy We do use mockall, but I'm not sure how you plan to use it. Here is one of our use cases, and the only one. :sob: https://github.com/datafuselabs/databend/blob/c54f06fc3788307aad67511e494dece562cb6d67/common/management/tests/it/user.rs

Reply:
> Two methods:
> 1. add the test to src/tests , remove test = false.
> 2. Make it public for now and put it under tests/it.

Thanks, I'm pretty clear now.

> but I'm not sure how you plan to use it.

Think of two structs `A` and `B`, where `B` is one of the fields of `A`. In the unit test, where `#[cfg(test)]` is on ), I'll mock a `B` and inject it into A to test how `A` interacts with `B`. In the integration test, where `#[cfg(test)]` is off, I'll test `A` with the real `B`.

To give a concrete example:

```rust
#[cfg(test)]
use other::MockB as B;
#[not(cfg(test))]
use other::B;

pub struct A {
    inner: B
}

mod other {
    #[cfg_attr(test, mockall::automock)]
    pub struct B {
        // ...
    }
} 
```

Reply:
got it, thanks. It looks like our testing style is not up to the task for now.